frontend/src/api/BaseApi.ts

```diff
@@ -0,0 +1,151 @@
+import { compact } from "lodash";
+
+export type ApiMethod = "DELETE" | "GET" | "PATCH" | "POST" | "PUT";
+export interface JSONRequestBody {
+  [key: string]: unknown;
+}
+
+export interface ApiResponseBody<TResponseData> {
+  message: string;
+  data: TResponseData;
+  status_code: number;
+  errors?: unknown[]; // TODO: define error and warning Issue type
+  warnings?: unknown[];
+}
```

Comment:
This needs to be modified for the actual payload, this is just a stand-in for now